StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentMethodsView.swift

```diff
@@ -218,23 +217,31 @@ class EmbeddedPaymentMethodsView: UIView {
 
     // MARK: Mandate handling
     private func updateMandate(animated: Bool = true) {
-        self.mandateView.attributedText = mandateProvider.mandate(for: selection?.paymentMethodType,
-                                                                  savedPaymentMethod: selection?.savedPaymentMethod,
-                                                                  bottomNoticeAttributedString: nil)
-
+        let mandateText = mandateProvider.mandate(
+            for: selection?.paymentMethodType,
+            savedPaymentMethod: selection?.savedPaymentMethod,
+            bottomNoticeAttributedString: nil
+        )
+        let shouldDisplayMandate: Bool = {
+            guard let mandateText else {
+                return false
+            }
+            return shouldShowMandate && !mandateText.string.isEmpty
+        }()
+        mandateView.attributedText = mandateText
+        let updateMandateUI = {
+            let spacing = shouldDisplayMandate ? 12.0 : 0
+            let mandateViewIndex = self.stackView.arrangedSubviews.firstIndex(of: self.mandateView)! - 1
+            let subviewBeforeMandateView = self.stackView.arrangedSubviews[mandateViewIndex]
+            self.stackView.setCustomSpacing(spacing, after: subviewBeforeMandateView)
+            self.mandateView.setHiddenIfNecessary(!shouldDisplayMandate)
+        }
         guard animated else {
-            self.mandateView.setHiddenIfNecessary(
-                (self.mandateView.attributedText?.string.isEmpty ?? true) ||
-                !shouldShowMandate
-            )
+            updateMandateUI()
             return
         }
-
         UIView.animate(withDuration: 0.25, animations: {
-            self.mandateView.setHiddenIfNecessary(
-                (self.mandateView.attributedText?.string.isEmpty ?? true) ||
-                !self.shouldShowMandate
-            )
+            updateMandateUI()
             self.setNeedsLayout()
             self.layoutIfNeeded()
```

Comment:
I would have expected `setNeedsLayout` and `layoutIfNeeded` to be inside the `updateMandateUI()` closure?

Reply:
I think it makes more sense if `layoutIfNeeded` isn't called if animation = false so that callers can control when exactly the view is laid out.

Reply:
ah! makes sense!